core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/ConnectionServiceImpl.java

```diff
@@ -112,10 +115,24 @@ public void ensureDataAccessPermission() throws ConnectionServiceException {
   }
 
   public List<IDatabaseConnection> getConnections() throws ConnectionServiceException {
+    return getConnections( false );
+  }
+
+  public List<IDatabaseConnection> getConnections( boolean hidePassword ) throws ConnectionServiceException {
     ensureDataAccessPermission();
     List<IDatabaseConnection> connectionList = null;
     try {
       connectionList = datasourceMgmtSvc.getDatasources();
+      if ( hidePassword ) {
+        for ( IDatabaseConnection conn : connectionList ) {
+          sanitizer.unsanitizeConnectionParameters( conn );
+          conn.setPassword( null );
```

Comment:
Suggestion: have only one *for {}* block and do:
`if ( hidePassword ) {
  conn.setPassword( null );
}`